traffic_control/tests/test_barrier_api.py

```diff
@@ -1,11 +1,37 @@
 import datetime
 
+import pytest
 from django.urls import reverse
 from rest_framework import status
 
 from traffic_control.models import BarrierPlan, BarrierReal, ConnectionType, Reflective
 
-from .test_base_api import TrafficControlAPIBaseTestCase
+from .factories import get_api_client, get_barrier_plan, get_barrier_real
+from .test_base_api import (
+    line_location_test_data,
+    point_location_test_data,
+    TrafficControlAPIBaseTestCase,
+)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    "location,query_location,expected",
+    [*point_location_test_data, *line_location_test_data],
+)
+def test_filter_barrier_plans_location(location, query_location, expected):
+    """
+    Ensure that filtering with location is working correctly.
+    """
+    api_client = get_api_client()
+
+    get_barrier_plan(location)
+    response = api_client.get(
+        reverse("api:barrierplan-list"), {"location": query_location.ewkt}
+    )
+
+    assert response.status_code == status.HTTP_200_OK
+    assert response.data.get("count") == expected
```

Comment:
Maybe better to validate also the response contents somehow, like check if it's empty list or none when geometry query does not return anything.

Reply:
Added ID checking for the response, which should cover everything in my opinion.